runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5253,15 +5258,28 @@ class INTERPRETER_CLASS
 		_currentThread->callOutCount += 1;
 #endif /* JAVA_SPEC_VERSION >= 19 */
 		updateVMStruct(REGISTER_ARGS);
-		VM_VMAccess::inlineExitVMToJNI(_currentThread);
+#if JAVA_SPEC_VERSION >= 21
+		/* Only exit to JNI for non-trivial downcalls for better performance. */
+		if (!_currentThread->isInTrivialDownCall)
+#else /* JAVA_SPEC_VERSION >= 21 */
+		{
+			VM_VMAccess::inlineExitVMToJNI(_currentThread);
+		}
+#endif /* JAVA_SPEC_VERSION >= 21 */
```

Comment:
I don't think this is what you intended (`#else` should be `#endif`):
```c
#endif /* JAVA_SPEC_VERSION >= 21 */
		{
			VM_VMAccess::inlineExitVMToJNI(_currentThread);
		}
```

Reply:
Fixed.